packages/circuits/src/gm_encryption.rs

```diff
@@ -0,0 +1,173 @@
+            expected_ciphertext_assigned.iter().zip(actual_ciphertext_assigned.iter())
+        {
+            ctx.constrain_equal(expected, actual);
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs::File;
+
+    use anyhow::{anyhow, Context, Ok, Result};
+    use halo2_ecc::fields::FpStrategy;
+    use halo2_wasm::{CircuitConfig, Halo2Wasm};
+    use probabilistic_encryption::{goldwasser_micali, key::PublicKey};
+    use serde::{Deserialize, Serialize};
+
+    use super::{GmVerificationCircuit, GmVerificationInputs};
+
+    #[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+    pub struct CircuitParams {
+        strategy: FpStrategy,
+        degree: u32,
+        num_advice: usize,
+        num_lookup_advice: usize,
+        num_fixed: usize,
+        lookup_bits: usize,
+        limb_bits: usize,
+        num_limbs: usize,
+    }
+
+    fn mock_gm_encryption() -> Result<GmVerificationInputs> {
+        let plaintext = b"hello world";
+
+        let (gm_pk, gm_sk) =
+            goldwasser_micali::generate_keys(8).expect("Failed to generate GM keys!");
+
+        let ciphertext = gm_pk.encrypt(plaintext);
+
+        Ok(GmVerificationInputs { gm_pk, gm_sk, ciphertext, plaintext: plaintext.to_vec() })
+    }
+
+    #[test]
+    fn test_gm_decryption_mock() -> Result<()> {
+        let path = "configs/gm_encryption.config";
+        let circuit_params: CircuitConfig = serde_json::from_reader(
+            File::open(path)
+                .map_err(|e| anyhow!(e))
+                .with_context(|| format!("The circuit config file does not exist: {}", path))?,
+        )
+        .map_err(|e| anyhow!(e))
+        .with_context(|| format!("Failed to read the circuit config file: {}", path))?;
+
+        let gm_verification_inputs = mock_gm_encryption()?;
+
+        let mut halo2_wasm = Halo2Wasm::new();
+
+        halo2_wasm.config(circuit_params);
+
+        let mut circuit = GmVerificationCircuit::new(&halo2_wasm, gm_verification_inputs);
+
+        circuit.verify_decryption();
+
+        halo2_wasm.mock();
+
+        Ok(())
+    }
+
+    // #[test]
+    // fn test_gm_encryption_mock() -> Result<()> {
+    //     let path = "configs/gm_encryption.config";
+    //     let circuit_params: CircuitConfig = serde_json::from_reader(
+    //         File::open(path)
+    //             .map_err(|e| anyhow!(e))
+    //             .with_context(|| format!("The circuit config file does not exist: {}", path))?,
+    //     )
+    //     .map_err(|e| anyhow!(e))
+    //     .with_context(|| format!("Failed to read the circuit config file: {}", path))?;
+
+    //     let gm_verification_inputs = mock_gm_encryption()?;
+
+    //     let mut halo2_wasm = Halo2Wasm::new();
+
+    //     halo2_wasm.config(circuit_params);
+
+    //     let mut circuit = GmVerificationCircuit::new(&halo2_wasm, gm_verification_inputs);
+
+    //     circuit.verify_encryption();
+
+    //     halo2_wasm.mock();
+
+    //     Ok(())
+    // }
```

Comment:
This snippet seems to be the same as the one above it? Was it left by mistake or we are keeping it for a reason?

Reply:
No it wasn't left by mistake, I was thinking how to get this test successful at the end as the expected constraint is that the `expected_cipher` is not equal to the `actual_cipher`. But I didn't find a way for doing such a constraint yet. So I will remove this test unit for now.